package.json

```diff
@@ -39,30 +39,32 @@
   "devDependencies": {
     "@biomejs/biome": "^1.9.4",
     "@rspack/core": "1.1.4",
-    "@types/node": "^22.10.1",
     "@types/jest": "29.5.14",
-    "react-refresh": "^0.14.2",
+    "@types/node": "^22.10.1",
     "cross-env": "^7.0.3",
     "execa": "9.5.1",
     "fs-extra": "11.2.0",
     "jest": "29.7.0",
     "jest-cli": "29.7.0",
     "jest-environment-node": "29.7.0",
     "nano-staged": "^0.8.0",
+    "react-refresh": "^0.14.2",
     "semver": "7.6.3",
     "simple-git-hooks": "^2.11.1",
     "ts-jest": "29.2.5",
+    "ts-node": "^10.9.2",
     "typescript": "5.7.2"
   },
   "dependencies": {
     "error-stack-parser": "^2.1.4",
     "html-entities": "^2.5.2"
   },
   "peerDependencies": {
-    "react-refresh": ">=0.10.0 <1.0.0"
+    "react-refresh": ">=0.10.0 <1.0.0",
+    "webpack-hot-middleware": "2.x"
   },
   "peerDependenciesMeta": {
-    "react-refresh": {
```

Comment:
I was kind of confounded by this being an optional dependency.

Correct me if i'm wrong, @LingyuCoder, but this plugin can't work if this dependency doesn't exist.

Right?

Reply:
cc @ahabhgk , since `react-refresh` in `react-refresh-webpack-plugin` is not optional, is there any special reason why it was set as optional?